src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -44,33 +44,72 @@ public AnnotatedOperation(string name, MethodInfo handler, ISwaggerModelCatalog
                 .Select(CreateSwaggerResponseObject)
                 .ToDictionary(x => x.GetStatusCode().ToString(), y => (global::Swagger.ObjectModel.Response) y);
 
-            var infos = handler.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any()).ToList();
+            var paramsList = new List<Parameter>();
+            //first check if any RouteParamAttributes are on the Method.
+            CreateSwaggerParametersFromMethodAttributes(handler, paramsList);
```

Comment:
Rather than creating a list and passing it in - I think you should have these functions return a list and then you can do `list.AddRange( ... )` to combine them.

Reply:
Why would that be any better? In this case, to detect duplicates or an existing Body param I would still need to pass in the list to the second method. Creating a single list performs slightly better too,  in a large application this could add up.